### PR TITLE
Buyers Guide campaign page

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Related PRs/issues: #9082

This PR introduces the following:
- A new buyers guide page model, `BuyersGuideCampaignPage`, which is very similar to the original CampaignPage model, with some uneccesary fields removed.
- A new relation model `BuyersGuideCamppaignPageModalRelation`, which allows users to link DonationModals to the page.
- Accompanying migrations for both of the models
- New Factories for both models
- A test file `test_campaign_page.py` which tests:
  - Model Factories
  - Basic Page Functionality
  - The BuyersGuideCampaignPage's `get_donation_modal_json` method
- New HTML template `buyersguide/campaign_page.html`
- Updated some petition and CTA related  JS/CSS files from being Foundation site specific, to now being share dbetween foundation and the buyersguide

# Steps to test 

1. Check out this branch and run `inv new-db` in the command line
2. 
3. Navigate to the buyers guide article index page in the CMS, and note that there are new `BuyersGuideCampaignPage`s at the bottom of the list.
4. Edit an existing page, or create a new one from scratch.
5. The CMS should work as expected, allowing you to select a CTA, donation modals if desired, and streamfields for the body.
6. Visit the page on the front end and confirm the page data is being displayed
7. Visit the Editorial Content Index page and confirm the Campaign pages are being rendered in the list (Refined styling to be taken care of in future ticket)
8. If everything works as expected, testing is complete! 

# Screenshots 

Campaign Page Template:
Screenshot 2022-10-17 at 23-59-22 Privacy Not Included A Buyer’s Guide for Connected Products

Campaign Page in Editorial Content Index list:
Screenshot 2022-10-17 at 23-59-31 Privacy Not Included A Buyer’s Guide for Connected Products

# Checklist

**Tests**
- [x] Is the code I'm adding covered by tests?

**Changes in Models:**
- [x] Did I update or add new fake data?
- ~~[ ] Did I squash my migration?~~
- ~~[ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+
+class BuyersGuideCampaignPage(FoundationMetadataPageMixin, Page):
+    parent_page_types = ['BuyersGuideEditorialContentIndexPage']
+    subpage_types: list = []
+    template = 'pages/buyersguide/campaign_page.html'
+
+    cta = models.ForeignKey(
+        'CTA',
+        related_name='buyersguide_campaign_page_for_cta',
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:49]: Can you add some context in what case the backward relation would be used?

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                     ('video', customblocks.VideoBlock()),
+                     ('linkbutton', customblocks.LinkButtonBlock()),
+                     ('looping_video', customblocks.LoopingVideoBlock()),
+                     ('pulse_listing', customblocks.PulseProjectList()),
+                     ('single_quote', customblocks.SingleQuoteBlock()),
+                     ('slider', customblocks.FoundationSliderBlock()),
+                     ('spacer', customblocks.BootstrapSpacerBlock()),
+                     ('airtable', customblocks.AirTableBlock()),
+                     ('datawrapper', customblocks.DatawrapperBlock()),
+                     ('typeform', customblocks.TypeformBlock()), )
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:85]: What do you think about unindenting these so they are only indented one more level than the wrapping parenthesis?

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        # Content tab fields
+
+        # FIXME: Contingency fix while https://github.com/mozilla/foundation.mozilla.org/pull/7771 is sorted out
+        # TranslatableField('cta'),
+        TranslatableField('title'),
+        TranslatableField('header'),
+        SynchronizedField('narrowed_page_content'),
+        TranslatableField('body'),
+        TranslatableField('donation_modal_relations'),
+    ]
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:125]: What do you think about listing the content fields before the settings fields? I feel like that would be more consistent with the Wagtail UI and how you listed the panels above.

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                     ('spacer', customblocks.BootstrapSpacerBlock()),
+                     ('airtable', customblocks.AirTableBlock()),
+                     ('datawrapper', customblocks.DatawrapperBlock()),
+                     ('typeform', customblocks.TypeformBlock()), )
+    )
+
+    def get_donation_modal_json(self):
+        modals = self.donation_modal_relations.all()
+        modals_json = [m.donation_modal.to_simple_dict() for m in modals]
+        return json.dumps(modals_json)
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:91]: What do you think about creating a utility function for this? I guess the same code is needed on the non-buyerguide donation page and the format is required by the JS? I know it's not the third repetition of this pattern but I guess the dependent JS might make it worth it.

File: network-api/networkapi/wagtailpages/tests/buyersguide/test_campaign_page.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+class FactoriesTest(test_base.WagtailpagesTestCase):
+    def test_campaign_page_factory(self):
+        buyersguide_factories.BuyersGuideCampaignPageFactory()
+
+    def test_campaign_page_donation_modal_relation_factory(self):
+        buyersguide_factories.BuyersGuideCampaignPageDonationModalRelationFactory()
+
+
+class BuyersGuideCampaignPageTest(test_base.WagtailpagesTestCase):
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/tests/buyersguide/test_campaign_page.py:18]: **Kudos** for the testing! 👍

**Review by tbrlpld — COMMENTED** [review]
Thanks @danielfmiranda. This is coming along nicely. Love the test coverage 👍 I have left a few minor question.

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        help_text='Choose existing or create new donation modal'
+    )
+
+    panels = [
+        SnippetChooserPanel('donation_modal'),
+    ]
+
+    class Meta(TranslatableMixin.Meta, Orderable.Meta):
+        verbose_name = 'Donation Modals'
+        verbose_name_plural = 'Donation Modals'
```
[the remark this answers is not in the document]
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py]: Hi Tibor, good catch! 

You are correct that this is not being used anywhere, so I opted to just use `pass`, similar to all the other models that inherit from `TranslatableMixIn` and `Orderable`


```
    class Meta(TranslatableMixin.Meta, Orderable.Meta):
        pass
```

File: network-api/networkapi/wagtailpages/tests/buyersguide/test_campaign_page.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        """
+        Testing the campaign pages "get_donation_modal_json" method
+        with no modals set. Should return empty JSON list.
+        """
+        campaign_page = buyersguide_factories.BuyersGuideCampaignPageFactory(
+            parent=self.content_index
+        )
+        list_returned_from_method = campaign_page.get_donation_modal_json()
+
+        self.assertEqual(list_returned_from_method, '[]')
```
[the remark this answers is not in the document]
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/tests/buyersguide/test_campaign_page.py]: This makes sense to me! I have updated the file. 👍

File: network-api/networkapi/wagtailpages/tests/buyersguide/test_editorial_content_index.py [hunk 8]
```diff
@@ -91,7 +91,10 @@ def test_parents(self):
     def test_children(self):
         self.assertAllowedSubpageTypes(
             parent_model=pagemodels.BuyersGuideEditorialContentIndexPage,
-            child_models={pagemodels.BuyersGuideArticlePage},
+            child_models={
+                pagemodels.BuyersGuideArticlePage,
+                pagemodels.BuyersGuideCampaignPage
+                },
```
[the remark this answers is not in the document]
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/tests/buyersguide/test_editorial_content_index.py]: Updated!

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py (hunk 2 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py]: Updated the formatting

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py (hunk 3 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py]: This makes sense to me as well, updated in latest commit!

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py (hunk 4 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:92]: Hi Tibor, I would vote against this as there is a minor difference between the functions for `CampaignPage`, and `BuyersGuideCampaignPage`.

For `BuyersGuideCampaignPage`, I was able to update the relationship to now use an orderable, similar to what we use for other recent models.

So in this function the name of the relationship has been updated from `donation_modals` to `donation_modal_relations` and line 90 has been updated from 

```
        modals_json = [m.to_simple_dict() for m in modals]
``` 
to

```
        modals_json = [m.donation_modal.to_simple_dict() for m in modals]
```

File: source/js/common/inject-react/index.js [hunk 9]
```diff
@@ -7,4 +8,5 @@ import injectJoinUs from "./join-us.js";
  */
 export const injectCommonReactComponents = (apps, siteUrl) => {
   injectJoinUs(apps, siteUrl);
+  injectPetition(apps, siteUrl);
```
[the remark this answers is not in the document]
Comment (tbrlpld) [reply source/js/common/inject-react/index.js:11]: Thanks @danielfmiranda. So if I understand correctly "foundation" JS does not end up being available in the buyersguide?

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py (hunk 1 above)
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:48]: The forward relation makes sense of course, from the page to the CTA. But do we ever need to go from the CTA to a page?

File: network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py (hunk 4 above)
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/pagemodels/buyersguide/campaign_page.py:92]: Thanks for expanding on this. I think the difference stems from the fact that the original relation model `DonationModals` also defines that `to_simple_dict` method... which in a weird way defers to the `DonationModeal.to_simple_dict`...

**Review by tbrlpld — APPROVED** [review]
Thanks @danielfmiranda. Looks good. 

Please remember to prefix the  branch names with the ticket numbers for future reference.

**Comment by tbrlpld** [comment]
The positioning of the does not work for me by the way. Not sure how to run the frontend tooling here to make that work. 


Screenshot 2022-10-19 at 14-51-00 Privacy Not Included A Buyer’s Guide for Connected Products

**Review by fessehaye — APPROVED** [review]

**Pull request merged**